src/plugins/newsfeed/public/plugin.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as Rx from 'rxjs';
+import { catchError, takeUntil } from 'rxjs/operators';
+import ReactDOM from 'react-dom';
+import React from 'react';
+import { I18nProvider } from '@kbn/i18n/react';
+import { PluginInitializerContext, CoreSetup, CoreStart, Plugin } from 'src/core/public';
+import { NewsfeedPluginInjectedConfig } from '../types';
+import { NewsfeedNavButton } from './components/newsfeed_header_nav_button';
+import { getApi } from './lib/api';
+
+export type Setup = void;
+export type Start = void;
+
+export class NewsfeedPublicPlugin implements Plugin<Setup, Start> {
+  private readonly kibanaVersion: string;
+  private readonly stop$ = new Rx.ReplaySubject(1);
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.kibanaVersion = initializerContext.env.packageInfo.version;
+  }
+
+  public setup(core: CoreSetup): Setup {}
+
+  public start(core: CoreStart): Start {
+    function mount(targetDomElement: HTMLElement) {
+      ReactDOM.render(
+        <I18nProvider>
+          <NewsfeedNavButton apiFetchResult={api$} />
+        </I18nProvider>,
+        targetDomElement
+      );
+      return () => ReactDOM.unmountComponentAtNode(targetDomElement);
+    }
+
+    core.chrome.navControls.registerRight({
+      order: 1000,
+      mount,
+    });
+
+    const { http, injectedMetadata } = core;
+    const config = injectedMetadata.getInjectedVar(
+      'newsfeed'
+    ) as NewsfeedPluginInjectedConfig['newsfeed'];
+
+    const api$ = getApi(http, config, this.kibanaVersion).pipe(
+      takeUntil(this.stop$), // stop the interval when stop method is called
+      catchError(() => {
+        // show a message to try again later?
+        // do not throw error
+        return Rx.of(null);
+      })
+    );
+  }
+
```

Comment:
This is subjective, and not a blocker, but I'm not crazy about the use of hoisting here to pull `api$` up, as it's easy to miss how it's being used.
Can we perhaps extract that into a method or inline it?

For the first few seconds it just looked like we weren't using `api$` at all, and I feel that makes maintenance a little harder by adding to the cognitive load.